helm/templates/deployment.yaml

```
@@ -32,6 +32,10 @@ spec:
               {{- toYaml . | nindent 12 }}
             {{- end }}
           env:
+            {{- if .Values.metadata.cluster }}
+            - name: KUBERNETES_CLUSTER_NAME
+              value: "{{ .Values.metadata.cluster }}"
+            {{- end -}}
```

Nit: for consistency `{{- end }}` as chomping whitespaces left is good enough.
```suggestion
            {{- with .Values.metadata.cluster }}
            - name: KUBERNETES_CLUSTER_NAME
              value: "{{ . }}"
            {{- end }}
```